packages/dd-trace/src/plugins/database.js

```diff
@@ -1,9 +1,63 @@
 'use strict'
 
 const StoragePlugin = require('./storage')
+const TextMapPropagator = require('../opentracing/propagation/text_map.js')
+const Config = require('../../src/config')
+const config = new Config()
+const textMap = new TextMapPropagator(config)
+const carrier = {}
 
 class DatabasePlugin extends StoragePlugin {
   static get operation () { return 'query' }
+
+  constructor (...args) {
+    super(...args)
+    this.serviceTags = {
+      dddbs: '',
+      encodedDddbs: '',
+      dde: '',
+      encodedDde: '',
+      ddps: '',
+      encodedDdps: '',
+      ddpv: '',
+      encodedDdpv: ''
+    }
+  }
+
+  createDBMPropagationCommentService () {
+    if (this.serviceTags.dddbs !== this.config.service) {
+      this.serviceTags.dddbs = this.config.service
+      this.serviceTags.encodedDddbs = encodeURIComponent(this.config.service)
+    }
+    if (this.serviceTags.dde !== this.tracer._env) {
+      this.serviceTags.dde = this.tracer._env
+      this.serviceTags.encodedDde = encodeURIComponent(this.tracer._env)
+    }
+    if (this.serviceTags.ddps !== this.tracer._service) {
+      this.serviceTags.ddps = this.tracer._service
+      this.serviceTags.encodedDdps = encodeURIComponent(this.tracer._service)
+    }
+    if (this.serviceTags.ddpv !== this.tracer._version) {
+      this.serviceTags.ddpv = this.tracer._version
+      this.serviceTags.encodedDdpv = encodeURIComponent(this.tracer._version)
+    }
+
+    return `dddbs='${this.serviceTags.encodedDddbs}',dde='${this.serviceTags.encodedDde}',` +
+    `ddps='${this.serviceTags.encodedDdps}',ddpv='${this.serviceTags.encodedDdpv}'`
+  }
+
+  injectDbmQuery (query) {
+    if (this.config.dbmPropagationMode === 'service') {
+      const servicePropagation = this.createDBMPropagationCommentService()
+      return `/*${servicePropagation}*/ ${query}`
+    } else if (this.config.dbmPropagationMode === 'full') {
+      config.experimental.traceparent = true
+      const servicePropagation = this.createDBMPropagationCommentService()
```

Comment:
This is repeated in both branches of the condition.

Reply:
@rochdev I discussed this with Thomas as well, I put them under the conditionals of this functions because leaving at the top means it is called regardless if propagation is enabled or not. Currently the function (`injectDbmQuery`) is called in the plugin without any conditions needing to be met.

Reply:
@tlhunter @rochdev @bengl anything to update here?

Reply:
@crysmags One way to handle this is to early-return if propagation is disabled. This default is `none` right? so you could do something like:

```js
const mode = this.config.dbmPropagationMode
if (mode === 'none') {
  return query
}
const servicePropagation = this.createDBMPropagationCommentService()
if (mode === 'service') {
  return `/*${servicePropagation}*/ ${query}`
} else if (mode === 'full') {
  config.experimental.traceparent = true
  textMap._injectTraceparent(this.activeSpan._spanContext, carrier)
  return `/*${servicePropagation},traceparent='${carrier.traceparent}'*/ ${query}`
} else {
  // This shouldn't happen because mode can only be 'none', 'service', or 'full', but
  // in case it's something else, we'll return the query
  return query
}

Reply:
got it, updated this function